extension/basicauth/README.md

```diff
@@ -0,0 +1,39 @@
+# Basic Authenticator
+
+This extension implements `configauth.ServerAuthenticator` to authenticate clients using HTTP Basic Authentication. The authenticator type has to be set to `basicauth`.
+
+## Configuration
+
+The following settings are available:
+
+- `htpasswd` - Path to the htpasswd file used to store the credentials (defaults to `.htpasswd`).
+
+### Example
+
+```yaml
+extensions:
+  basicauth:
+    htpasswd: .htpasswd
```

Comment:
Should this also accept username / password fields to reference environment variables (and hopefully not embedded directly though it could happen...)? Using extra files like `.htpasswd` can be complicated in environments like k8s

Reply:
I am not sure how mounting a single file can be complicated, but I can also add an option to inline `.htpasswd` content into the config and supply it through env vars.

Reply:
I believe that was the idea of @jpkrohling from https://github.com/open-telemetry/opentelemetry-collector/issues/1424.

Reply:
I think we should keep using a file first, as this is what I would expect from an extension like this. If we have requirements from users to embed the contents of the file in the config, we can consider adding that.

Reply:
I see that you implemented this already. This is probably fine, I'll review it once @anuraaga gives his approval.